diff --git a/Android.bp b/Android.bp
index cce5062d19672768ac9a5646133d0416d56f9b92..037d8311a3bd462fcedad8ebf271be6a0f08b38b 100644
--- a/Android.bp
+++ b/Android.bp
@@ -16,6 +16,7 @@ license {
 cc_library {
     name: "zuid_md5",
     host_supported: true,
+    vendor_available: true,
     sdk_version: "current",
     stl: "c++_shared",
     target: {
@@ -37,6 +38,7 @@ cc_library {
 cc_library {
     name: "xmp_toolkit_sdk",
     host_supported: true,
+    vendor_available: true,
     sdk_version: "current",
     stl: "c++_shared",
     export_include_dirs: [
